Alignment/OfflineValidation/plugins/EopElecTreeWriter.cc

```diff
@@ -0,0 +1,915 @@
+  const reco::SuperClusterCollection* EndcapSupClusCollection = &iEvent.get(theEndCapSupClusCollectionToken_);
+
+  // necessary to re-calculate phi and eta width of SuperClusters
+  SuperClusterShapeAlgo SCShape(rhc, subGeo);
+
+  //---------------    Trigger   -----------------
+  TrigTag = false;
+  const edm::TriggerResults* trigRes = &iEvent.get(theTriggerResultsToken_);
+
+  // trigger event
+  const trigger::TriggerEvent* triggerEvent = &iEvent.get(theTriggerEventToken_);
+
+  // our trigger table
+  std::map<std::string, EopTriggerType> HLTpaths;
+  for (unsigned int i = 0; i < triggerNames_.size(); i++) {
+    if (triggerNames_[i].find(theTrigger_) != 0)
+      continue;
+    EopTriggerType myTrigger;
+
+    const unsigned int prescaleSize = hltConfig_.prescaleSize();
+    for (unsigned int ps = 0; ps < prescaleSize; ps++) {
+      const unsigned int prescaleValue = hltConfig_.prescaleValue(ps, triggerNames_[i]);
+      if (prescaleValue != 1) {
+        myTrigger.prescale = prescaleValue;
+      }
+    }
+
+    myTrigger.index = hltConfig_.triggerIndex(triggerNames_[i]);
+    if (myTrigger.index == -1)
+      continue;
+    myTrigger.fired =
+        trigRes->wasrun(myTrigger.index) && trigRes->accept(myTrigger.index) && !trigRes->error(myTrigger.index);
+    HLTpaths[triggerNames_[i]] = myTrigger;
+  }
+
+  // First cut : trigger cut
+  std::string firstFiredPath = "";
+  for (const auto& it : HLTpaths) {
+    if (it.second.fired) {
+      TrigTag = true;
+      firstFiredPath = it.first;
+      break;
+    }
+  }
+  if (!TrigTag)
+    return;
+  h_nEventsTriggered->Fill(0.5);
+
+  // Displaying filters label from the first fired trigger
+  // Useful for finding the good filter label
+  std::vector<std::string> filters = hltConfig_.moduleLabels(firstFiredPath);
+
+  if (debugTriggerSelection_) {
+    edm::LogInfo("EopElecTreeWriter") << "filters : ";
+    for (unsigned int i = 0; i < filters.size(); i++) {
+      edm::LogInfo("EopElecTreeWriter") << filters[i] << " ";
+    }
+  }
+
+  // Getting HLT electrons
+  edm::InputTag testTag(theFilter_, "", "HLT");
+  int testindex = triggerEvent->filterIndex(testTag);
+
+  if (testindex >= triggerEvent->sizeFilters())
+    return;
+  h_nEventsHLTFilter->Fill(0.5);
+
+  const trigger::Keys& KEYS_el(triggerEvent->filterKeys(testindex));
+
+  std::vector<const trigger::TriggerObject*> HLTelectrons;
+  for (unsigned int i = 0; i != KEYS_el.size(); ++i) {
+    const trigger::TriggerObject* triggerObject_el = &(triggerEvent->getObjects().at(KEYS_el[i]));
+    HLTelectrons.push_back(triggerObject_el);
+  }
+
+  h_nHLTelectrons->Fill(HLTelectrons.size());
+
+  if (HLTelectrons.empty())
+    return;
+  h_nEventsHLTelectron->Fill(0.5);
+
+  // finding the HLT electron with highest pt and saving the corresponding index
+  unsigned int HighPtIndex = 0;
+  double maxPtHLT = -5.;
+  for (unsigned int j = 0; j < HLTelectrons.size(); j++) {
```

Comment:
range based loop?

Reply:
the index is used to determine `HighPtIndex` which is then used later on.